nav2_graceful_motion_controller/src/graceful_motion_controller.cpp

```diff
@@ -0,0 +1,344 @@
+// Copyright (c) 2023 Alberto J. Tudela Roldán
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_core/controller_exceptions.hpp"
+#include "nav2_util/geometry_utils.hpp"
+#include "nav2_graceful_motion_controller/graceful_motion_controller.hpp"
+#include "nav2_costmap_2d/costmap_filters/filter_values.hpp"
+
+namespace nav2_graceful_motion_controller
+{
+
+void GracefulMotionController::configure(
+  const rclcpp_lifecycle::LifecycleNode::WeakPtr & parent,
+  std::string name, const std::shared_ptr<tf2_ros::Buffer> tf,
+  const std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros)
+{
+  auto node = parent.lock();
+  parent_node_ = parent;
+  if (!node) {
+    throw nav2_core::ControllerException("Unable to lock node!");
+  }
+
+  costmap_ros_ = costmap_ros;
+  tf_buffer_ = tf;
+  plugin_name_ = name;
+  logger_ = node->get_logger();
+
+  // Handles storage and dynamic configuration of parameters.
+  // Returns pointer to data current param settings.
+  param_handler_ = std::make_unique<ParameterHandler>(
+    node, plugin_name_, logger_,
+    costmap_ros_->getCostmap()->getSizeInMetersX());
+  params_ = param_handler_->getParams();
```

Comment:
I think this causes a problem if the dynamic parameters are changing `params_` without a lock on this parameter pointer's getting of that memory. Maybe it makes sense for `getParams()` to return a parameters copy and the `getParams()` method is locked by the mutex. That way, the get/sets are mutually exclusive and the params can't change mid-execution here. We get a new `getParams()` at the start of each cycle. 

Though, that complicates anything initialized here in the configure method. I think there's a bit of a design problem that might need more than a simple solution to resolve